src/Processors/QueryPlan/AggregatingStep.cpp

```diff
@@ -46,6 +47,7 @@ static ITransformingStep::Traits getTraits(bool should_produce_results_in_order_
             .returns_single_stream = should_produce_results_in_order_of_bucket_number,
             .preserves_number_of_streams = false,
             .preserves_sorting = false,
+            .preserves_data_hints = false,
```

Comment:
in general I think it should be
1. if constrained column is present in the output header then `true`. otherwise - `false`.
2. if additionally the given step executes some actions there shouldn't be an `ALIAS` node with `result_name == constrained_col_name`